app/api/tax.py

```diff
@@ -57,24 +72,13 @@ def query(self, view_kwargs):
             query_ = query_.join(Event).filter(Event.identifier == event.id)
         return query_
 
-    def before_create_object(self, data, view_kwargs):
-        event = None
-        if view_kwargs.get('event_id'):
-            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
-        elif view_kwargs.get('event_identifier'):
-            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
-        if event:
-            data['event_id'] = event.id
-
     view_kwargs = True
-    decorators = (jwt_required, )
```

Comment:
Say for example you pay for an event XYZ, say it is said organized by Group ABC which is actually run by a company MNO. You will never see the name for MNO, their address or any such detail. You will know you have paid someone in Group ABC and get an email ID to contact in case of refund and all. You never know the underlying registered company. 
AFAIK. And as far as I have seen. @niranjan94 might be able to explain better.